### PR TITLE
[ros2] Adding GPS plugin

### PR DESCRIPTION
**Pull request by alexfneves**
I created the GPS sensor plugin that outputs a NavSatFix message in ros2. I used the IMU plugin as reference for the implementation.

I also added a test. First the box position acquired, then its position is set on the world (positive x, y and z), then check if the latitude, longitude and altitude are positive (positive East, North and Up).

Please take a look if it seems fine.

I have some notes:
1. P.S. Gazebo seems to be outputting ENU (the default world orientation) wrong. My take is that the C++ code shouldn't change, what should be fixed is the Gazebo code. I added a 180 deg on the heading on the .world file to make it output what it should. Later I'll take a look at the gazebo code to check the reason for this behavior.

2. The SDF has <position_sensing> and <velocity_sensing> tags, but I only implemented the <position_sensing> because the GpsSensor in Gazebo API for Gazebo 9.0 doesn't have the functions VelocityEast(), VelocityNorth() and VelocityUp(), which are available in Gazebo 9.1 and forth.

### CODE REVIEW

**Comment by alexfneves** [comment]
The test that is failing is the GazeboRosAckermannDriveTest Publishing, which is not related to the GPS.
http://build.ros2.org/job/Dpr__gazebo_ros_pkgs__ubuntu_bionic_amd64/105/testReport/

File: gazebo_plugins/include/gazebo_plugins/gazebo_ros_gps_sensor.hpp [hunk 1]
```diff
@@ -0,0 +1,66 @@
+// Copyright 2018 Open Source Robotics Foundation, Inc.
```
Comment (chapulina) [inline gazebo_plugins/include/gazebo_plugins/gazebo_ros_gps_sensor.hpp]: Nitpick: 2019. Also on the other files.

File: gazebo_plugins/src/gazebo_ros_gps_sensor.cpp [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+{
+}
+
+void GazeboRosGpsSensor::Load(gazebo::sensors::SensorPtr _sensor, sdf::ElementPtr _sdf)
+{
+  impl_->ros_node_ = gazebo_ros::Node::Get(_sdf);
+
+  impl_->sensor_ = std::dynamic_pointer_cast<gazebo::sensors::GpsSensor>(_sensor);
+  if (!impl_->sensor_) {
+    RCLCPP_ERROR(impl_->ros_node_->get_logger(), "Parent is not an gps sensor. Exiting.");
```
Comment (chapulina) [inline gazebo_plugins/src/gazebo_ros_gps_sensor.cpp]: Nitpick: "an gps" -> "a GPS"

File: gazebo_plugins/test/worlds/gazebo_ros_gps_sensor.world [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    <include>
+      <uri>model://ground_plane</uri>
+    </include>
+    <include>
+      <uri>model://sun</uri>
+    </include>
+    <spherical_coordinates>
+        <!-- currently gazebo has a bug: instead of outputing lat, long, altitude in ENU as the
+        default configurations, it's outputting (-E)(-N)U, therefore we rotate the default frame
+        180 so that it would go back to 180 -->
```
Comment (chapulina) [inline gazebo_plugins/test/worlds/gazebo_ros_gps_sensor.world]: Mind adding a link to this issue here so we don't lose track of it?

https://bitbucket.org/osrf/gazebo/issues/2022/

**Review by chapulina — CHANGES_REQUESTED** [review]
Thanks for the PR! It's looking good to me! Besides the nitpick comments, all I'd like to ask is that you create a demo world in the `gazebo_plugins/worlds` folder. It can be similar to the test world, but with a few instructions on the top. Thanks!

**Review by chapulina — APPROVED** [review]
Thanks for iterating, LGTM! :+1:

File: gazebo_plugins/test/worlds/gazebo_ros_gps_sensor.world (hunk 3 above)
Comment (alexfneves) [reply gazebo_plugins/test/worlds/gazebo_ros_gps_sensor.world]: I didn't search about it before. Good to know there is already an issue on this!

Comment (alexfneves) [reply gazebo_plugins/test/worlds/gazebo_ros_gps_sensor.world]: Sorry @chapulina, at the time I was making the changes I didn't understand that you wanted me to add this link https://bitbucket.org/osrf/gazebo/issues/2022/ on the .world comment. Now that it has been merged into master, maybe you can add it directly.

Comment (chapulina) [reply gazebo_plugins/test/worlds/gazebo_ros_gps_sensor.world]: No worries, it can be added later :wink:

**Pull request merged**